classes/web_processor.php

```diff
@@ -28,20 +28,27 @@
  */
 class web_processor implements processor {
 
-    /**
-     * @var profile $profile The profile object for the run.
-     */
+    /** @var profile $profile The profile object for the run. */
     protected $profile;
 
+    /** @var sample_set */
+    protected $sampleset;
+
     /**
      * Initialises the processor
      *
      * @param manager $manager The profiler manager object
      */
     public function init(manager $manager) {
+        $this->sampleset = new sample_set(
+            script_metadata::get_request(),
+            (int) $manager->get_starttime(),
+            script_metadata::get_samplelimit()
+        );
+
         $this->profile = new profile();
-        $this->profile->add_env(script_metadata::get_request());
-        $this->profile->set('created', (int) $manager->get_starttime());
+        $this->profile->add_env($this->sampleset->name);
+        $this->profile->set('created', $this->sampleset->starttime);
 
         $manager->get_timer()->setCallback(function($s) use ($manager) {
```

Comment:
Not exactly in this patch but since it's in view of the diff, can you please change the anonymous function to have a space after the function keyword please? This makes it consistent with the PSR standards (which Moodle defers to when not specified directly)

https://www.php-fig.org/psr/psr-12/#47-method-and-function-calls